dashboard/app/controllers/reports_controller.rb

```diff
@@ -29,6 +30,28 @@ def header_stats
     render file: 'shared/_user_stats', layout: false, locals: {user: current_user}
   end
 
+  def user_progress
```

Comment:
This is my new route at /popup/progress
Please review and let me know if I should be throwing different errors.